### ISSUE TITLE
Review mode disabled because GraphQL support not discovered for GHE

### ISSUE THREAD

**Issue by ChrisHines** [microsoft/vscode-pull-request-github#947]
Issue Type: <b>Bug</b>

GitHub Enterprise 2.15.3

1. Open a workspace with a GHE repo. (In case it's relevant, the local repo has a `git@<host>:<user>/<repo>.git` style remote.)
2. Open the output panel
3. View Github Pull Request output

I get:
```
[Info] Looking for git repository
[Info] Git repository found, initializing review manager and pr tree view.
[Info] Found GitHub remote
[Info] https://<redacted>/api/v3: GraphQL not supported (Network error: Response not successful: Received status code 404)
[Info] Review> no matching pull request metadata found for current branch master
```
I expected graphql to work. Clicking through the API doc links on our instance of GHE brings me to this page https://developer.github.com/enterprise/2.15/v4/guides/intro-to-graphql/. Following that documentation I found that the following command works using the same token that I have configured for the vscode-github-pr extention.
```
curl -H "Authorization: bearer <redacted>" https://<redacted>/api/graphql
```
The following also works, but returns a much smaller JSON response with a series of URL patterns that doesn't look like a GraphQL result.
```
curl -H "Authorization: bearer <redacted>" https://<redacted>/api/v3
```

The extension does not enable the batch review mode feature because it cannot find the GraphQL support.

Extension version: 0.4.0
VS Code version: Code 1.31.0 (7c66f58312b48ed8ca4e387ebd9ffe9605332caa, 2019-02-05T22:32:14.164Z)
OS version: Darwin x64 17.7.0

**Comment by tyrsius** [comment]
I'm also experiencing this bug with this exact error, but oddly several of my teammates are able to use this plugin with the same GHE instance. I think more logging needs to be added to help identify the issue.

**Comment by StanleyGoldman** [comment]
It seems the endpoint a graphql query was being sent to is incorrect
https://developer.github.com/enterprise/2.15/v4/guides/forming-calls/#the-graphql-endpoint

The endpoint should take the form of...
> http(s)://[hostname]/api/graphql

**Comment by StanleyGoldman** [comment]
Hey @ChrisHines I'm trying real hard to reproduce your issue but I'm not having much luck.
I initially cloned via HTTPS, then via SSH, and both times the pull request list loads, the detail page displays.

I'm using a 2.16 instance of GHE and I'm going to get my hands on a 2.15 instance and test again.

I was hoping if you could try again with the latest version of the plugin.

@tyrsius Can you let us know what version of GitHub Enterprise you are testing against?

**Comment by ChrisHines** [comment]
@StanleyGoldman I just tried again. The problem still exists. We still have GHE 2.15.3 here, but my VSCode is now 1.32.1 and my GitHub Pull Requests version is now 0.5.0.

### Clarification

The extension _mostly_ works. I can see the list of pull requests. I can load the detail page, and I can checkout the PR. But I **cannot** do batch reviews. When I click in the editor to add a review comment, I cannot see a `Start Review` button. I only see the `Add Comment` button. My understanding was that review mode should be implemented based on #546 being closed.

When it didn't work I looked at the two commits linked at the bottom of #546 and it looked to me like the review mode functionality is [gated by support for graphql](https://github.com/Microsoft/vscode-pull-request-github/commit/cd7f73ae8c6f7a339337e1f9b85823375f131ef2#diff-899893b79f6b773695a02418b37bb9b5R1002). That's why I mentioned in my initial issue post above that it seemed the extension was having problems finding the graphql endpoint.

**Comment by tyrsius** [comment]
I believe this is caused by PR names that fail to deserialize. Possibly from length, possibly from having emoji in them.

**Comment by StanleyGoldman** [comment]
Thanks for the clarification, I'm able to reproduce this...

On a public GitHub repo with a PR not created by me:
![image](https://user-images.githubusercontent.com/417571/56300731-7f004b80-6104-11e9-92f7-8e73633c841a.png)

On a private GitHub Enterprise instance with a PR not created by me:
![image](https://user-images.githubusercontent.com/417571/56300606-42345480-6104-11e9-96b3-4a2c56a5893c.png)

**Comment by sreenath1491** [comment]
I am having the same problem here. Unable to see the `Start Review` button and see the error message `https://<redacted>/api/v3: GraphQL not supported (Network error: Response not successful: Received status code 404)`. I did a little debugging with vscode insiders and found the issue with url used for graphql at this line: https://github.com/microsoft/vscode-pull-request-github/blob/master/src/github/credentials.ts#L191. The `baseUrl=https://<redacted>/api/v3` and the link function here: https://github.com/microsoft/vscode-pull-request-github/blob/master/src/github/credentials.ts#L274 seems to append `/graphql` to the url making the final graphQL url: `https://<redacted>/api/v3/graphql` thus giving a 404 response. It should be `https://<redacted>/api/graphql`. I was able to get a successful response by changing the URL. I am not sure if the cause has already been found .

**Comment by ChrisHines** [comment]
I would really like to use the review mode feature on my GHE projects. Is there anything I can do to help get a fix for this issue into the next release? @StanleyGoldman has reproduced the issue and @sreenath1491 has done some initial debugging here. What is the next step?

**Comment by Cammisuli** [comment]
There's also some changes that are required here:
https://github.com/microsoft/vscode-pull-request-github/blob/f883c462bd5884cb2ba131d8a0f811949f74fd90/src/authentication/configuration.ts#L20-L24

**Comment by ChrisHines** [comment]
@StanleyGoldman This issue is assigned to you, are you still working on it?

**Comment by ChrisHines** [comment]
This was fixed for me by #1382 in release 0.13.0.
